src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -6047,6 +6048,38 @@ class CreateRoleStatementSegment(ansi.CreateRoleStatementSegment):
     )
 
 
+class CreateDatabaseRoleStatementSegment(BaseSegment):
+    """A `CREATE DATABASE ROLE` statement.
+
+    https://docs.snowflake.com/en/sql-reference/sql/create-database-role
+    """
+
+    type = "create_database_role_statement"
+    match_grammar = Sequence(
+        "CREATE",
+        Sequence(
+            "OR",
+            "REPLACE",
```

Comment:
Let's do `OrReplaceGrammar`

Reply:
Commit https://github.com/sqlfluff/sqlfluff/pull/5475/commits/8ff17623b38f77f5c5ec347a3e6e33c460ad12d9 contains this change. Thanks for the pointer!

Would it be helpful if I filed an issue in this project to track the tech debt of converting other uses of `Sequence("OR", "REPLACE", ...` to `Ref("OrReplaceGrammar")`?

Reply:
Could be an interesting issue. I wonder if there's some way to automatically flag such things in our CI pipeline.